numpy/linalg/linalg.py

```diff
@@ -2112,6 +2112,9 @@ def norm(x, ord=None, axis=None, keepdims=False):
     """
     x = asarray(x)
 
+    if not issubclass(x.dtype.type, floating):
```

Comment:
I don't think this is the right check. Typically, we handle this by adding `0.0` to the result, which preserves inexact types and respects complex. Note
```
In [10]: a = array([1,2,3], complex)

In [11]: issubclass(a.dtype.type, floating)
Out[11]: False
```
You can also do
```
In [13]: issubdtype(a.dtype, np.inexact)
Out[13]: True
```